cores/esp32/esp32-hal-rgb-led.c

```diff
@@ -1,3 +1,6 @@
+#include "soc/soc_caps.h"
+#if SOC_RMT_SUPPORTED
+
 #include "esp32-hal-rgb-led.h"
```

Comment:
Note, since arduino-esp32 can use managed components now: we can use [led_strip](https://components.espressif.com/components/espressif/led_strip) component here, which has API-compatible implementations based on RMT and SPI. Then the feature will be still available for C2, and there would be less code to maintain.

Reply:
If you are suggesting to replace our RGB code with led_strip, then this would be a larger item than this PR should handle, because it involves Peripheral Manager. I would suggest to leave it as-is and note in the PR that we need to add led_strip support (or figure out something else that would make sense)

Reply:
My goal was to do as less as changes as possible. The suggested change is great imho should be done after this PR has merged.

Reply:
My suggestion would be to wrap the contents of the function instead. You can even `log_e` in `#else` to prompt us that this function is not implemented yet.

Reply:
added log_e

Reply:
Like this:
```cpp
void neopixelWrite(uint8_t pin, uint8_t red_val, uint8_t green_val, uint8_t blue_val){
#if SOC_RMT_SUPPORTED
    // ... the current code
#else
    log_e("RMT is not supported on " CONFIG_IDF_TARGET);
#endif /* SOC_RMT_SUPPORTED */
}
```

Reply:
Yes, this comment wasn't intended to be a PR blocker. I just asked to consider this and create a follow-up task if you think this is a good idea to consider.

Reply:
One of the main reasons Peripheral Manager was implemented is to keep track of used peripherals and allow us to pick an available one that can achieve the requested functionality. Ex. PWM can be done with LedC, RMT, etc. Same can be said for neopixels (RMT, I2S, SPI, etc.). Probably proper task would be to implement such functionality and expand the options for all chips

Reply:
For sure there should be a follow up here, to provide an universal working solution for every MCU, choosen by default the "best" solution (with user override, when set).